src/crypto_utils.ts

```diff
@@ -150,35 +181,112 @@ export async function buildStringInputText(
 
     let encodedStr = encoder.encode(plaintext)
 
-    let encryptedStr = new Array<{ ciphertext: bigint, signature: Buffer }>(plaintext.length)
+    let encryptedStr = new Array<{ ciphertext: bigint, signature: Uint8Array }>(plaintext.length)
 
     for (let i = 0; i < plaintext.length; i++) {
-        const {ctInt, signature} = buildInputText(BigInt(encodedStr[i]), sender, contractAddress, functionSelector)
-        encryptedStr[i] = {ciphertext: ctInt, signature}
+        const { ctInt, signature } = buildInputText(BigInt(encodedStr[i]), sender, contractAddress, functionSelector)
+        encryptedStr[i] = { ciphertext: ctInt, signature }
     }
 
     return encryptedStr
 }
 
-export function generateAesKey() {
-    return crypto.randomBytes(block_size).toString("hex")
+export function decryptUint(ciphertext: bigint, userKey: string): bigint {
+    // Convert ciphertext to Uint8Array
+    let ctArray = new Uint8Array()
+
+    let temp = new Uint8Array()
```

Comment:
why to create it outside of the loop?

Reply:
I don't think there is any need to re-init the variable on every iteration of the loop. Perhaps this is the Solidity developer in me that feels this way